snuba/datasets/discover.py

```diff
@@ -111,6 +113,35 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
         source.set_table_source(detected_dataset)
 
 
+class SimpleColumnValidator(QueryProcessor):
+    """
+    Checks that requested columns are present on the Discover schema. Prevents a column
+    that happens to be present on an underlying dataset being inadvertently exposed
+    on Discover.
+    """
+
+    def __init__(self, all_columns: ColumnSet) -> None:
+        self.__columns = all_columns
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        aliases = query.get_aliases()
+        columns = [
+            col for col in query.get_all_referenced_columns() if col not in aliases
+        ]
+        time_group_columns = [
+            "time",
+            "bucketed_end",
```

Comment:
Should discover be using bucketed_end? I thought the time alias was going to be the bucketed column.

Reply:
No, we need to make "time" work for both datasets. Sooon.....

Reply:
Good to know that `time` isn't ready yet. :+1:

Reply:
Is `time` column being used anywhere for Discover v2?

Reply:
Yes we use it in the timeseries endpoints, and it is an exposed column in the public schema.